### ISSUE TITLE
Unable to fetch values from AWS SecretsManager

### ISSUE THREAD

**Issue by doron-cohen** [hashicorp/packer#9792]
When filing a bug, please include the following headings if possible. Any
example text in this template can be deleted.

#### Overview of the Issue

I am trying to load secrets from aws_secretsmanager.
I have an AWS credentials file that assumes role to another account:
```
[dev]
role_arn = arn:aws:iam::************:role/******
source_profile = default
```
and in the AWS config file I set:
```
[profile dev]
region = eu-central-1
```
When running `packer build` with `AWS_PROFILE=dev` I get the following error:
```
Error initializing core: template: root:1:3: executing "root" at <aws_secretsmanager `/infrastructure/tlv/vm_root_user_password`>: error calling aws_secretsmanager: Error getting secret: MissingRegion: could not find region configuration
```
I followed #9237 advice and also set `AWS_SDK_LOAD_CONFIG=true`.
This seems to solve the problem but now I get what looks like a parsing error (?):
```
Error initializing core: template: root:1:3: executing "root" at <aws_secretsmanager `/infrastructure/tlv/vm_root_user_password`>: error calling aws_secretsmanager: Error getting secret: invalid character 'k' looking for beginning of value
```
The secret plaintext does start with a `k` but why is that a problem?

#### Reproduction Steps

I don't know if the profile settings are related to the issue. It looks like it is able to read the response from the AWS API but there is a bug in the parsing of it. But it is hard for me to believe that this mechanism is totally broken.

### Packer version

Packer v1.6.0

### Simplified Packer Buildfile

```json
{
  "variables": {
    "vcenter_server": "{{ env `VCENTER_SERVER` }}",
    "vcenter_username": "{{ env `VCENTER_USERNAME` }}",
    "vcenter_password": "{{ aws_secretsmanager `/infrastructure/tlv/vcenter_password` }}",
    "ssh_root_password": "{{ aws_secretsmanager `/infrastructure/tlv/vm_root_user_password` }}"
  },
  "builders": [
    {
      "type": "vsphere-iso",

      "vcenter_server": "{{ user `vcenter_server` }}",
      "username": "{{ user `vcenter_username` }}",
      "password": "{{ user `vcenter_password` }}",
...
```

### Operating system and Environment details

```
Linux workbench 5.7.2-arch1-1 #1 SMP PREEMPT Wed, 10 Jun 2020 20:36:24 +0000 x86_64 GNU/Linux
```

**Comment by doron-cohen** [comment]
Upgrading to Packer 1.6.1 doesn't help. Only formats the error differently:

```
Error: 

template: root:1:3: executing "root" at <aws_secretsmanager
`/infrastructure/tlv/vm_root_user_password`>: error calling aws_secretsmanager:
Error getting secret: invalid character 'k' looking for beginning of value
```

**Comment by nywilken** [comment]
Hello @doron-cohen thanks for reaching out. I believe you are running into two separate issues (#9771 and #9237 ) which have both been resolved on master and are set to be released in the upcoming v1.6.2 release. In the meantime, you can download the latest [nightly release](https://github.com/hashicorp/packer/releases/tag/nightly) which has the fix for both issues. 

Seeing as these issues have been resolved already I am going to close this issue. But if you find that you are still running into issues or find a new bug please leave a comment on this thread and I will gladly reopen.
